lib/block-supports/layout.php

```diff
@@ -65,7 +66,7 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 		$style .= "$selector .alignleft { float: left; margin-right: 2em; margin-left: 0; }";
 		$style .= "$selector .alignright { float: right; margin-left: 2em; margin-right: 0; }";
 		if ( $has_block_gap_support ) {
-			$gap_style = $gap_value ? $gap_value : 'var( --wp--style--block-gap )';
+			$gap_style = $gap_value && ! $should_skip_gap_serialization ? $gap_value : 'var( --wp--style--block-gap )';
```

Comment:
Yeah, this is another tricky one. 🤣 

I agree in principle, and it did cross my mind, however there's a consideration that caused me to hesitate: `--wp--style--block-gap`'s value comes from `theme.json`.

See: https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/themes/theme-json.md#what-is-blockgap-and-how-can-i-use-it

So my reasoning at the time was, since `skipSerialization` acts at the block level, and is indeed meant to offer [finer-grained control of block supports](https://github.com/WordPress/gutenberg/issues/28913), it probably shouldn't override the theme's settings. 

Skipping serialization, as far as I understand it, skips user-facing block customization, that is the editor styles controls. 

Aside from all that, `blockGap` is another one of those anomalies. It straddles block supports and layout, and doesn't make life easy when working with both.

Hopefully we can consolidate/homogenize the use of layout and blockGap in the effort to make styles and layouts consistent. See https://github.com/WordPress/gutenberg/issues/39336

Reply:
Yeah this is a tough one. On the whole I'm more inclined to leave it out because the theme only defines the variable value, not where to use it, and I'd expect skipSerialization to not output _anything_. But I think we can leave it as is and revisit later; outputting a couple of default styles shouldn't be a huge deal.